internal/handlers/common/update.go

```diff
@@ -622,6 +700,57 @@ func extractValueFromUpdateOperator(op string, update *types.Document) (*types.D
 		return nil, NewWriteErrorMsg(ErrFailedToParse, "Modifiers operate on fields but we found another type instead")
 	}
 
+	if op == "$rename" {
+		iter := doc.Iterator()
+		keys := map[string]struct{}{}
+
+		for {
+			k, v, err := iter.Next()
+			if err != nil {
+				if errors.Is(err, iterator.ErrIteratorDone) {
+					break
+				}
+
+				return nil, err
```

Comment:
Let's return a lazyerror here as, in principle, we don't expect to have an error here, and it would be nice to have it wrapped so we can follow the stack of functions being called.

Reply:
This part of code now is removed so I guess it's resolved